stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -693,6 +697,20 @@ end
     end
 end
 
+Base._reverse(A::Tridiagonal, dims) = reverse!(Matrix(A); dims)
+Base._reverse(A::Tridiagonal, dims::Colon) = Tridiagonal(reverse(A.du), reverse(A.d), reverse(A.dl))
+function Base._reverse!(A::Tridiagonal, dims::Colon)
+    n = length(A.du) # == length(A.dl), & always 1-based
+    n == length(A.dl) || throw(DimensionMismatch("Diagonals have different lengths"))
```

Comment:
```suggestion
```
This check shouldn't be required since it is enforced by the `Tridiagonal` constructor.

Reply:
Normally, yes. But when using `@inbounds` the criteria are stricter:

https://github.com/JuliaLang/julia/pull/54231#discussion_r1579615903

Reply:
I disagree, as commented above.